src/main/io/dataEdition.c

```diff
@@ -0,0 +1,1359 @@
+static void applyNewValuesOnCleanflightDataPage(int8_t whichWay)
+{
+	switch (selectedItem) {
+		case 3 :{
+			if ((valueTable[structureOfPage.editField[3].data].type & VALUE_MODE_MASK) == MODE_LOOKUP){
+				structureOfPage.editField[9].data = *(uint8_t *)valueTable[structureOfPage.editField[3].data].ptr;
+				structureOfPage.editField[9].minValue = 0;
+				structureOfPage.editField[9].maxValue = lookupTables[valueTable[structureOfPage.editField[3].data].config.lookup.tableIndex].valueCount - 1;
+			}
+			break;
+		}
+		case 9 :{
+			int_float_value_t tmp;
+			tmp.int_value = structureOfPage.editField[9].data; // case of lookup mode
+			if ((valueTable[structureOfPage.editField[3].data].type & VALUE_MODE_MASK) == MODE_DIRECT){
+				if ((valueTable[structureOfPage.editField[3].data].type & VALUE_TYPE_MASK) == VAR_FLOAT){
+					valueFloat = constrainf((valueFloat + (whichWay * 0.001f)), valueTable[structureOfPage.editField[3].data].config.minmax.min, valueTable[structureOfPage.editField[3].data].config.minmax.max);
+					tmp.float_value = valueFloat;
+				}
+				else {
+					valueInteger = constrain((valueInteger + whichWay), valueTable[structureOfPage.editField[3].data].config.minmax.min, valueTable[structureOfPage.editField[3].data].config.minmax.max);
+					tmp.int_value = valueInteger;
+				}
+			}
+			cliSetVar(&valueTable[structureOfPage.editField[3].data], tmp);
+			break;
+		}
+	}
+}
+
+#if (DATA_EDITION_FULL_VERSION)
+static void testAndPositionThatBit(uint8_t testValue, uint32_t bitMask)
+{
+	if (testValue == 1)
+	    featureSet(bitMask);
+	else
+	    featureClear(bitMask);
+}
+
+static void applyNewValuesOnThatPage(void)
+{
+	uint8_t item, index;
+
+    switch (selectedPage){
+	    case FEATURES_PAGE : {
+	        index = (whichOfPages * DISPLAY_LINES_MINUS2) + (selectedLine-1);
+	    	testAndPositionThatBit(structureOfPage.editField[selectedItem].data, (1 << index));
+	        break;
+	    }
+        case MEASURES_PAGE : {
+        	isAlarmSoundOn = structureOfPage.editField[15].data;
+        	masterConfig.telemetryConfig.hottAlarmSoundInterval = structureOfPage.editField[18].data;
+		    break;
+		}
+        case BATTERY_PAGE : {
+	    	testAndPositionThatBit(structureOfPage.editField[3].data, FEATURE_VBAT);
+			masterConfig.batteryConfig.vbatmincellvoltage = structureOfPage.editField[6].data;
+			masterConfig.batteryConfig.vbatmaxcellvoltage = structureOfPage.editField[9].data;
+			masterConfig.batteryConfig.vbatwarningcellvoltage = structureOfPage.editField[12].data;
+			masterConfig.batteryConfig.vbatscale = structureOfPage.editField[15].data;
+
+			batteryWarningVoltage = batteryCellCount * masterConfig.batteryConfig.vbatwarningcellvoltage;
+			batteryCriticalVoltage = batteryCellCount * masterConfig.batteryConfig.vbatmincellvoltage;
+		    break;
+		}
+        case CURRENTMETER_PAGE : {
+	    	testAndPositionThatBit(structureOfPage.editField[3].data, FEATURE_CURRENT_METER);
```

Comment:
This is the effect of my initial choice (only int8 in the menu structure).
If I'm right (see *sensors/battery.c* line 168), this was probably more than enough for this kind of measurement and dynamic variable as  :
    - offset : a step of 10 represents a step of 0.25A with a scale factor of 400 (commonly used 40mV/A)
    - gain : a step of 10 leads to an error of around 2.5%.

Anyway, data values are now int32 (see other comment) which results in a higher precision.